packages/devtools/kit/src/promise.ts

```diff
@@ -0,0 +1,150 @@
+import type { Path } from './valtio';
+
+export type PromiseResolve<T> = (value: T | PromiseLike<T>) => void;
+export type PromiseReject = (reason?: any) => void;
+
+export const isPromiseLike = <T = any>(obj: unknown): obj is PromiseLike<T> =>
+  obj !== null &&
+  typeof obj === 'object' &&
+  typeof (obj as any).then === 'function';
+
+const _promiseResolveMapping = new WeakMap<Promise<any>, PromiseResolve<any>>();
```

Comment:
Do we need to clean the map at certain times ?

Reply:
Since the resolve & reject methods all inhert from corresponding promise. I think use WeakMap to store relations of them would be memory-safe.